wagtail/wagtailadmin/static_src/wagtailadmin/js/page-editor.js

```diff
@@ -315,9 +315,9 @@ function InlinePanel(opts) {
 
 function cleanForSlug(val, useURLify) {
     if (URLify != undefined && useURLify !== false) { // Check to be sure that URLify function exists, and that we want to use it.
-        return URLify(val);
+        return URLify(val, true);
```

Comment:
This needs to be `URLify(val, null, true)` - the second parameter is `num_chars`. Also, it looks like we need to import `js/vendor/xregexp/xregexp.min.js` from Django admin when unicode slugs are in use.